pysteps/downscaling/rainfarm.py

```diff
@@ -44,7 +44,60 @@ def _balanced_spatial_average(x, k):
     return convolve(x, k) / convolve(ones, k)
 
 
-def downscale(precip, ds_factor, alpha=None, threshold=None, return_alpha=False):
+def _smoothconv(P,nas):
+
+    """
+    Parameters
+    ----------
+    
+    P: matrix
+    matrix with the input field to smoothen, with dimensions ns*ns
+    
+    nas : int
+    original size
```

Comment:
why do you need an argument for the original size? Can't you just get it from `P`?

Reply:
I get this `nas` on the low resolution field (the field to be reduced) on the `downscale()` method. The `P` used here already has a high resolution.